src/control/tools/EditSelection.cpp

```diff
@@ -407,136 +414,73 @@ void EditSelection::mouseMove(double mouseX, double mouseY, bool alt) {
-                f = (this->snappedBounds.height - (snappedY - this->snappedBounds.y)) / this->snappedBounds.height;
-                f = (this->height * f < minSize) ? 1 : f;
-                this->scaleShift(1, f, false, true);
-            }
+            ySide = -1;
         } else if (this->mouseDownType == CURSOR_SELECTION_BOTTOM) {
-            yOffset -= this->height;
-            f = std::max((this->height + yOffset) / this->height, minSize / this->height);
-            this->scaleShift(1, f, false, false);
-
-            if (this->snappedBounds.height > minSize) {
-                double snappedY =
-                        snappingHandler.snapVertically(this->snappedBounds.y + this->snappedBounds.height, alt);
-                f = (this->snappedBounds.height + (snappedY - this->snappedBounds.y - this->snappedBounds.height)) /
-                    this->snappedBounds.height;
-                f = (this->height * f < minSize) ? 1 : f;
-                this->scaleShift(1, f, false, false);
-            }
+            ySide = 1;
         } else if (this->mouseDownType == CURSOR_SELECTION_LEFT) {
-            f = std::max((this->width - xOffset) / this->width, minSize / this->width);
-            this->scaleShift(f, 1, true, false);
-
-            if (this->snappedBounds.width > minSize) {
-                double snappedX = snappingHandler.snapHorizontally(this->snappedBounds.x, alt);
-                f = (this->snappedBounds.width - (snappedX - this->snappedBounds.x)) / this->snappedBounds.width;
-                f = (this->width * f < minSize) ? 1 : f;
-                this->scaleShift(f, 1, true, false);
-            }
+            xSide = -1;
         } else if (this->mouseDownType == CURSOR_SELECTION_RIGHT) {
-            xOffset -= this->width;
-            f = std::max((this->width + xOffset) / this->width, minSize / this->width);
-            this->scaleShift(f, 1, false, false);
-
-            if (this->snappedBounds.width > minSize) {
+            xSide = 1;
+        }
+        // sanity check
+        if (xSide || ySide) {
+            // get normalized direction vector for input interpretation (dependent on aspect ratio)
+            double diag = hypot(xSide * this->width, ySide * this->height);
+            double nx = xSide * this->width / diag;
+            double ny = ySide * this->height / diag;
+
+            int xMul = (xSide + 1) / 2;
+            int yMul = (ySide + 1) / 2;
+            double xOffset =
+                    (rx - this->x) - this->width * xMul;  // x-offset from corner/side that is used for resizing
+            double yOffset =
+                    (ry - this->y) - this->height * yMul;  // y-offset from corner/side that is used for resizing
+
+            // calculate scale factor using dot product
+            double f = (xOffset * nx + yOffset * ny + diag) / diag;
+            f = std::copysign(std::max(std::abs(f), minSize / std::min(this->width, this->height)), f);
```

Comment:
I thinks this needs correction. You will see the difference if your selection consists of an ellipse with a large eccentricity.
```suggestion
            f = std::copysign(std::max(std::abs(f), minSize / std::min(std::abs(this->width), std::abs(this->height))), f);
```